product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs

```diff
@@ -13,8 +13,10 @@ namespace roundhouse.infrastructure.filesystem
     /// <summary>
     /// All file system access code comes through here
     /// </summary>
-    public sealed class WindowsFileSystemAccess : FileSystemAccess
+    public sealed class DotNetFileSystemAccess : FileSystemAccess
     {
+        private static readonly bool is_running_on_mono = Type.GetType("Mono.Runtime") != null;

```

Comment:
Also need add DbProviderFactories section in app.config
``` xml
 <system.data>
        <DbProviderFactories>
          <remove invariant="MySql.Data.MySqlClient" />
          <add name="MySQL Data Provider" invariant="MySql.Data.MySqlClient" description=".Net Framework Data Provider for MySQL" type="MySql.Data.MySqlClient.MySqlClientFactory, rh" />
        </DbProviderFactories>
    </system.data>
 ```

I'm not testing all features. Only simplest rh configuration and mysql provider..